router/batchrouter/handle.go

```diff
@@ -116,6 +116,9 @@ type Handle struct {
 	lastExecTimesMu sync.RWMutex
 	lastExecTimes   map[string]time.Time
 
+	failingDestinationsMu sync.RWMutex
+	failingDestinations   map[string]bool
```

Comment:
```suggestion
	failingDestinations   map[string]struct{}
```
Is it possible to use empty structs here, since currently it is being used as set only?

Reply:
Then we would need to delete from the set when the destination starts succeeding. Both are equally valid solutions in my head. I went with the one that requires the less amount of lines :)